### DISCUSSION TITLE
Modifiability - 2

### DISCUSSION THREAD

**Discussion by changjurhee** [changjurhee/AONE#17] [Polls]
## Quality Attributes

Type | Description | QA Scenario (Stimulus / Source of stiulus / Artifact / Environment / Response / Response measure) | Tatics
 -- | -- | -- | --
Modifiability | Provide the ability for the developer/maintainer to easily add or replace encryption algorithms. | Upon customer request, encryption algorithm change for data communication must be possible within 5 hours.<br>- Encryption algorithm change request<br>- Customer<br>- Encryption algorithm related software components<br>- During encryption algorithm change<br>- System supports the changed encryption algorithm<br>- It takes less than 5 hours to change the encryption algorithm |

**Comment by ju297** [comment]
Codec과 동일하게 취급하면 될 것 같습니다.

Layer Pattern

장점사용하는 layer와 사용 당하는 layer가 확실하게 구분되어 하위 레이어만 교체하는데 용이
Layer bridging을 한다면 변경 용이성을 달성할 수 없음

Pipes and Filters Pattern

데이터 스트림을 처리하기에 적합하며 교체 또한 용이
데이터 병렬처리에 대해서는 다소 부적합 할 수 있음

**Comment by codekistone** [comment]
split module - same with codec
